src/v/cluster/rm_stm.cc

```diff
@@ -2434,15 +2436,18 @@ ss::future<stm_snapshot> rm_stm::take_snapshot() {
       "Removing abort indexes {} with offset < {}",
       expired_abort_indexes.size(),
       start_offset);
-
-    for (const auto& idx : expired_abort_indexes) {
-        auto filename = abort_idx_name(idx.first, idx.last);
-        vlog(
-          _ctx_log.debug,
-          "removing aborted transactions {} snapshot file",
-          filename);
-        co_await _abort_snapshot_mgr.remove_snapshot(filename);
-    }
+    f = ss::do_with(
```

Comment:
mind adding a comment with reference to the compiler issue and mention not to coroutinze?